bare_metal_billing/models.py

```diff
@@ -0,0 +1,74 @@
+from decimal import Decimal
+from typing import Annotated
+from typing_extensions import Self
+import datetime
+
+import pydantic
+
+
+BM_UNKNOWN_SU = "BM UNKNOWN"
+
+
+def parse_date(v: str | datetime.date) -> datetime.date:
+    if isinstance(v, str):
+        return datetime.datetime.fromisoformat(v)
+    return v
+
+
+def parse_bm_accelerator(v: int | list) -> int:
+    """
+    The BM usage data obtained from ESI does not explicitly say how many GPUs a node has.
+    Instead, the number of GPUs is based on the number of entries in the `Resource Properties.accelerators` field
+    """
+    if isinstance(v, list):
+        return len(v)
+    return v
+
+
+PositiveDecimalField = Annotated[Decimal, pydantic.Field(gt=0)]
+PositiveIntField = Annotated[int, pydantic.Field(gt=-1)]
+DateField = Annotated[datetime.datetime, pydantic.BeforeValidator(parse_date)]
+
+
+class BMResourceProperties(pydantic.BaseModel):
+    cpu_count: Annotated[PositiveIntField, pydantic.Field(alias="cpus")] = 0
+    ram: Annotated[PositiveIntField, pydantic.Field(alias="memory_mb")] = 0
+    gpu_count: Annotated[
```

Comment:
in fact, why do we even care about how many cpu or ram a node has? Aren't we directly billing based on the resourceclass?